server/src/main/java/io/crate/planner/statement/CopyPlan.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.statement;
+
+import static io.crate.data.SentinelRow.SENTINEL;
+
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+
+import io.crate.data.CollectingRowConsumer;
+import io.crate.data.InMemoryBatchIterator;
+import io.crate.data.Row;
+import io.crate.data.Row1;
+import io.crate.data.RowConsumer;
+import io.crate.metadata.TransactionContext;
+
+
+public class CopyPlan {
```

Comment:
maybe we can make it abstract, make `CopyFromPlan` and `CopyToPlan` subclasses of this class, make execute protected and call it from children - not a big logical difference just kinda natural to expect that `CopyPlan` is the parent.

Reply:
@BaurzhanSakhariev I had it this way but see review comment of @mfussenegger above:

> No need to make CopyPlan abstract and use inheritance here to re-use a static method.

Reply:
Didn't see it - well, matter of taste probably - I stumbled over this during review, probably because of the naming - with static approach I would use some different name. No strong opinion here, fine as is.

Reply:
Please don't create inheritance if it can be avoided (see numerous post about "why inheritance is bad").
Also I wonder if it is really worth to create a dedicated class for just 1 static method shared between to similar classes. Maybe move to `CopyFromPlan` and reuse at `CopyToPlan`?
If you prefer to have this new class, make it `final` please.

Reply:
Inheritance can be used if there is an appropriate relationship between the different classes, but COPY FROM and COPY TO are rather separate - that they share a common name doesn't imply a bigger relationship then say COPY TO has with INSERT INTO.

Making a single static function accessible for multiple components is a bad use of inheritance because inheritance  couples components bi-directional.

In this particular case I think putting it in `CopyPlan` doesn't even fit, because the functionality to (not) wait for completion is not tied to COPY FROM or COPY TO either, but it is a general concept. It could even fit into the `JobLauncher`.
In some cases it is fine to put general functionality into concrete components if it isn't used anywhere else to keep the API surface smaller, but in this case we're even adding a new abstraction, so not sure if that point holds.